### PR TITLE
Ignore cleanup failures when applying ASG updates

### PR DESCRIPTION
**Pull request by mariash**
After new ASG chain is created and we fail to clean up old ASG chain we
log the error but continue on and consider our new ASG chain as
successful and save it in the in-memory map for subsequent comparison.
This is to avoid re-applying the same chain. As well as saving it in a
map as a desired chain to avoid deleting it right after we create it.

Original ASG chain fails to be deleted because cascading deletion of
referenced chains might fail. In the observed case we saw that
referenced chain is the netout-...-log chain that is getting referenced
by the new ASG chain now.

Signed-off-by: Marc Paquette <mpaquette@vmware.com>

### CODE REVIEW

File: src/code.cloudfoundry.org/vxlan-policy-agent/converger/converger_test.go [hunk 1]
```diff
@@ -603,48 +603,118 @@ var _ = Describe("Single Poll Cycle", func() {
...
 				// when cleaning up orphaned chains
-				i := 0
-				fakeEnforcer.EnforceRulesAndChainStub = func(e enforcer.RulesWithChain) (string, error) {
-					var err error
-					if i > 0 {
-						err = fmt.Errorf("eggplant")
-					}
-					i++
-					return fmt.Sprintf("%s-with-new-suffix", e.Chain.Prefix), err
-				}
```
Comment (geofffranks) [inline src/code.cloudfoundry.org/vxlan-policy-agent/converger/converger_test.go:614]: this looks like it was moved below, but the comment is now abandoned. should that get moved or cleaned up as well?

**Review by geofffranks — CHANGES_REQUESTED** [review]
Looks good aside from the orphaned comment

File: src/code.cloudfoundry.org/vxlan-policy-agent/converger/converger.go [hunk 2]
```diff
@@ -155,12 +155,14 @@ func (m *SinglePollCycle) SyncASGsForContainers(containers ...string) error {
 				})
 				chain, err := m.enforcer.EnforceRulesAndChain(ruleset)
 				if err != nil {
-					errors = multierror.Append(errors, fmt.Errorf("enforce-asg: %s", err))
+					if _, ok := err.(*enforcer.CleanupErr); ok {
+						m.logger.Error("failed-to-cleanup", err)
```
Comment (MarcPaquette) [inline src/code.cloudfoundry.org/vxlan-policy-agent/converger/converger.go:159]: Why don't we use `multierror.Append(errors, fmt.Errorf("failed-to-cleanup %s", err))` here?

Comment (MarcPaquette) [inline src/code.cloudfoundry.org/vxlan-policy-agent/converger/converger.go]: Why don't we use a `errors = multierror.Append(errors, fmt.Errorf("failed-to-cleanup: %s", err))` here?

File: src/code.cloudfoundry.org/vxlan-policy-agent/converger/converger_test.go (hunk 1 above)
Comment (mariash) [reply src/code.cloudfoundry.org/vxlan-policy-agent/converger/converger_test.go:614]: cool, thanks for the notice

**Review by MarcPaquette — APPROVED** [review]
LGTM

**Pull request merged**